Tests/AlicerceTests/Persistence/DiskMemoryPersistenceTestCase.swift

```diff
@@ -31,29 +31,15 @@ final class DiskMemoryPersistenceTestCase: XCTestCase {
         let expectation3 = self.expectation(description: "Save MrMinder3")
         persistMinder(with: "mr-minder2", into: diskMemPersistence, expectation: expectation3)
 
-        func checkAndRetry(times: UInt, execute: @escaping (@escaping () -> Void) -> Void) {
+        waitForExpectations(timeout: 1)
 
-            func retry() {
-                Thread.sleep(forTimeInterval: 5)
-                if times > 0 {
-                    checkAndRetry(times: times - 1, execute: execute)
-                }
-            }
-
-            execute(retry)
-        }
+        // wait for all operations (including the eviction ones) to finish
+        diskMemPersistence.writeOperationQueue.waitUntilAllOperationsAreFinished()
```

Comment:
i think we discussed at the time this was done to not expose properties just to make them usable by the tests. something that i don't agree and as far as i remember you neither.
well but ok 🤷‍♂️
that't why there was the 🔨 in the tests to validate to check if all the files have been written.

anyway, this is much nice way to validate, although it's breaking some "rules" (at least for me 😛 )

Reply:
It was mainly because of that test, since it was really slow and a bit weird, to be honest 😆

As for exposing these as `internal`, I know it's not "ideal", but also think that the impact should be minimal, since they won't be exposed to the framework's clients.

If you really are uncomfortable with this, we can try using a different approach to be able to inform whether all work has been completed without exposing the queues.